### ISSUE TITLE
Conflicting automatic template assignment settings

### ISSUE THREAD

**Issue by Ruslan-Aleev** [modxcms/revolution#16363]
### Summary
MODX has a new feature - template picker with preview. This is enabled by default and is set by the system setting `enable_template_picker_in_tree`.
BUT MODX also has settings for automatic template assignment - `automatic_template_assignment`. So, these settings contradict each other.

Why do I need to look at the template picker window and select a template every time I create a resource, if I have already set the template assignment in the `automatic_template_assignment` setting?

I propose to remove the `enable_template_picker_in_tree` setting, and make the template picker an option in `automatic_template_assignment`, now there are `system, parent and sibling` options, and it will become `system, parent, sibling and picker`.

As a result, there will be no conflicting and extra settings.

### Environment
MODX 3.x

**Comment by JoshuaLuckers** [comment]
The system setting `automatic_template_assignment` is used to determine the template that should be selected/used by default when creating a new resource. It doesn't restrict you from selecting another template.

The system setting `enable_template_picker_in_tree` is a restricting (should you be able to pick a template yes/no). 
What template the template picker selects by default depends on `automatic_template_assignment`. If that's not the case it's a bug in my opinion.

**Comment by Ruslan-Aleev** [comment]
> The system setting automatic_template_assignment is used to determine the template that should be selected/used by default when creating a new resource. It doesn't restrict you from selecting another template.

Yes, that's understandable. But if I set the setting, then it's easier for me to change the template for some resources than to select the template in the window every time.

> What template the template picker selects by default depends on automatic_template_assignment. If that's not the case it's a bug in my opinion.

Yes, it is also an option for the picker to take into account the `automatic_template_assignment` setting. Now the setting is not taken into account in picker.

**Comment by JoshuaLuckers** [comment]
To me the template picker behaves as expected, it not taking into account the `automatic_template_assigment` setting is a bug. 

But others should chime in as well so we can decide if this is actual a bug/feature in regards of how the template picker should work.

**Comment by smg6511** [comment]
I  can confirm that the template picker does not pick up the automatic assignments. I'd say that it should, as the picker is just a shortcut for previewing and specifying a Resource’s template ahead of editing its content.

Honestly though, I’ve found the template picker to be of limited usefulness and maybe, as @Ruslan-Aleev alludes to, it’s just an unnecessary extra step in creating a Resource. As the whole point of this feature seems to be the preview image, the move to make would have been to incorporate the preview image into the Quick Create Resource panel and perhaps make that preview available in the main editing page as well.

**Comment by Ruslan-Aleev** [comment]
Related with https://github.com/modxcms/revolution/issues/15933

**Comment by smg6511** [comment]
@Mark-H - Judging by your comments from #15933, you favor the template picker feature and maybe even use it. Given that, would you agree that it should respect/reflect the `automatic_template_assignment` / `default_template` settings? Anyone else looking at this thread, please chime in ... I don't want to embark on a _fix_ that is not in fact a fix for users who find the template picker useful ;-)

**Comment by Mark-H** [comment]
Yes, the new resource window (which is more than just a template picker - it also sets resource type and the title, which saves 2 refreshes when changing those without the new resource window) should ideally default to whatever those settings indicate should be used.

**Comment by smg6511** [comment]
Ok, then I'll look at fixing the issue at hand.

For the future, it might be worthwhile to rethink this workflow though. Now we have three avenues one can take to create a new Resource (quick create, template picker [when active], or the standard full create-edit), although only two of these are available at any given time. We can keep this down to two avenues and remove the extra template picker workflow by:

- adding the preview to the Quick Create window;
- adding a “Save, Continue Editing” button to the Quick Create window (choosing this action would essentially replicate the template picker flow);
- adding a system setting to toggle the default behavior of the Add/Create button in the trees between Standard and Quick Create; and finally,
- removing the template picker related logic and system setting.

**Comment by Ruslan-Aleev** [comment]
I also found a bug in the template picker - when creating a resource through it, the uniqueness of the aliases of the resources is NOT checked.

**Comment by modxcommunity** [comment]
This issue has been mentioned on **MODX Community**. There might be relevant details there:

https://community.modx.com/t/form-customization-issue-on-modx-3-0-5-pl/8013/3

**Comment by Ibochkarev** [comment]
Still open on 3.x. The two settings still do different jobs, and the template picker still ignores `automatic_template_assignment`.

`ResourceCreateManagerController::getDefaultTemplate()` in `manager/controllers/default/resource/create.class.php` switches on `automatic_template_assignment` (`system`, `parent`, `sibling`) and then applies a Form Customization `fieldDefault` for `template`. The tree does not use that when the picker is on. `createResourceHere` in `manager/assets/modext/widgets/resource/modx.tree.resource.js` calls `MODx.createResource()`, and `MODx.window.CreateResource` sets the template to `config.record.template || MODx.config.default_template` (`modx.window.resource.js`). Parent and sibling are skipped.

@JoshuaLuckers, @Mark-H, and @smg6511 already settled this: keep both settings. `enable_template_picker_in_tree` only chooses whether the tree opens the window. `automatic_template_assignment` chooses the default template. Turning the picker into a fourth value of that list would remove the window for people who want a default and a preview.

When the window opens, use the same default `getDefaultTemplate()` already computes.

- Add a connector, for example `Resource/GetDefaultTemplate`, taking `parent` and `context_key`. Share the method with the create page so the two paths cannot diverge.
- In `MODx.createResource` (`manager/assets/modext/core/modx.js`), request it before `show()` and pass the id as `record.template`.
- Leave `enable_template_picker_in_tree` as the on/off switch. Set it to No if the tree should go straight to the full create page, which already calls `getDefaultTemplate()`.

On the alias note from 2024: `core/src/Revolution/Processors/Resource/Create.php` does call `isDuplicateAlias`. The picker window only draws the alias field when friendly URLs are on and `automatic_alias` is off, and its `failure` listener only writes to the console. If that error is still invisible, it is a separate display bug.

#15933 is the older picker thread.
